client/my-sites/email/email-management/home/email-plan/index.jsx

```diff
@@ -121,19 +123,72 @@ function EmailPlan( { domain, hideHeaderCake = false, selectedSite, source } ) {
 		);
 	};
 
-	const addEmailForwardMutationActive = useAddEmailForwardMutationIsLoading();
+	// Email checkout provides the /mailbox route with a new email param, e.g. /mailboxes/example.com?new-email=example@example.com
+	const queryParams = new URLSearchParams( window.location.search );
+	const newEmail = queryParams.get( 'new-email' );
+
+	const refetchTimeoutRef = useRef( null );
+	const [ refetchCount, setRefetchCount ] = useState( 0 );
```

Comment:
Two nits:
1. Can we move this line closer to the `useGetEmailAccountsQuery` call?
2. Can we call it something more specific? E.g., `emailAccountsRefetchCount`.